cmd/dra-example-controller/driver.go

```diff
@@ -94,9 +94,17 @@ func (d driver) GetClaimParameters(ctx context.Context, claim *resourcev1.Resour
 
 func (d driver) Allocate(ctx context.Context, claim *resourcev1.ResourceClaim, claimParameters interface{}, class *resourcev1.ResourceClass, classParameters interface{}, selectedNode string) (*resourcev1.AllocationResult, error) {
 	if selectedNode == "" {
-		return nil, fmt.Errorf("TODO: immediate allocations not yet supported")
+		return d.allocateImmediateClaim(claim, claimParameters, class, classParameters)
 	}
 
+	if err := d.allocateDelayedClaim(claim, claimParameters, class, classParameters, selectedNode); err != nil {
+		return nil, err
+	}
+
+	return buildAllocationResult(selectedNode, true), nil
+}
+
+func (d driver) allocateDelayedClaim(claim *resourcev1.ResourceClaim, claimParameters interface{}, class *resourcev1.ResourceClass, classParameters interface{}, selectedNode string) error {
```

Comment:
It's inconsistent for this to only return an error, but `allocateImmediateClaim()` to return an AllocationResult and an error. They should probably both return an AllocationResult and an error. This should also make the body of the wrapping call simpler and more logical to follow.

Reply:
That's true. Both of them now return allocation result.